core/src/main/scala/org/apache/spark/api/r/RBackendHandler.scala

```diff
@@ -166,40 +178,75 @@ private[r] class RBackendHandler(server: RBackend)
 
   // Read a number of arguments from the data input stream
   def readArgs(numArgs: Int, dis: DataInputStream): Array[java.lang.Object] = {
-    (0 until numArgs).map { arg =>
+    (0 until numArgs).map { _ =>
       readObject(dis)
     }.toArray
   }
 
-  // Checks if the arguments passed in args matches the parameter types.
-  // NOTE: Currently we do exact match. We may add type conversions later.
+  // Find a matching method in all methods of the same name of a class
+  // according to the passed arguments. Arguments may be converted in
+  // order to match a method.
+  //
+  // Returns an Option[Int] which is the index of the matched method in
```

Comment:
Why does not return the found method directly, instead of index? Since we change the semantics of this function, we should also rename it, for example, findMethodByArgs ?